lib/api/apiUtils/object/corsResponse.js

```diff
@@ -0,0 +1,127 @@
+/** _matchesValue - compare two values to determine if they match
+* @param {string} allowedValue - an allowed value in a CORS rule;
+    may contain wildcards
+* @param {string} value - value from CORS request
+* @return {boolean} - true/false
+*/
+function _matchesValue(allowedValue, value) {
+    const wildcardIndex = allowedValue.indexOf('*');
+    // If no wildcards, simply return whether strings are equal
+    if (wildcardIndex === -1) {
+        return allowedValue === value;
+    }
+    // Otherwise make sure substrings match expected substrings before
+    // and after the wildcard
+    const beginValue = allowedValue.substring(0, wildcardIndex);
+    const endValue = allowedValue.substring(wildcardIndex + 1);
+    return (value.startsWith(beginValue) && value.endsWith(endValue));
+}
+
+/** _matchesOneOf - check if a value matches any values of a CORS rule array
+* @param {string[]} allowedValues - array of values allowed in CORS rule
+* @param {string} value - value from CORS request
+* @return {boolean} - true/false
+*/
+function _matchesOneOf(allowedValues, value) {
+    return allowedValues.some(allowedValue =>
+        _matchesValue(allowedValue, value));
+}
+
+/** _headersMatchRule - check if headers match AllowedHeaders of rule
+* @param {string[]} headers - expected headers of CORS request
+* @param {string[]} ruleAllowedHeaders - AllowedHeaders of a CORS rule
+* @return {boolean} - true/false
+*/
+function _headersMatchRule(headers, ruleAllowedHeaders) {
+    if (!ruleAllowedHeaders) {
+        return false;
+    }
+    // AllowedHeaders may have been stored with uppercase letters
+    // during putBucketCors; ignore case when searching for match
```

Comment:
This confused me too but they preserve case in putting / getting the cors resource. I can add some tests to substantiate this.